src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+                } else if (this->actor.colChkInfo.damageEffect == 3) {
+                    func_80876CAC(this);
+                    func_80878594(this);
+                } else {
+                    func_80876BD0(this, globalCtx, i);
+                    func_8087864C(this);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877DE0.s")
+void EnDodongo_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnDodongo* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877E60.s")
+    func_80878910(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 75.0f, 60.0f, 70.0f, 0x1D);
+    if (this->actor.bgCheckFlags & 2) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GERUDOFT_DOWN);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80878354.s")
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+    if (this->actionFunc != func_808787B0) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider2.base);
+        if ((this->actionFunc != func_808786C8) && (this->actionFunc != func_80877E60)) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+            if ((this->actionFunc != func_8087784C) && func_80877278(this, globalCtx)) {
+                func_80877DE0(this);
+            }
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80878424.s")
+    if (func_8087721C(this)) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider3.base);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80878594.s")
+    if (this->unk_33C > 0.0f) {
+        if (this->unk_300 != 10) {
+            Math_StepToF(&this->unk_33C, 0.0f, 0.05f);
+            this->unk_340 = (this->unk_33C + 1.0f) * 0.375f;
+            this->unk_340 = (this->unk_340 > 0.75f) ? 0.75f : this->unk_340;
+        } else if (!Math_StepToF(&this->unk_344, 0.75f, 0.01875f)) {
+            func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_808785B0.s")
+s32 func_80878E44(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDodongo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_8087864C.s")
+    if (limbIndex == 1) {
+        pos->z += 1000.0f;
+    } else if ((limbIndex == 15) || (limbIndex == 16)) {
+        Matrix_Scale(this->unk_314.x, this->unk_314.y, this->unk_314.z, MTXMODE_APPLY);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_808786C8.s")
+    return 0;
```

Comment:
```suggestion
    return false;
```
and they return true or false.